### PR TITLE
lists of inputs enforcing "required" constraint, and couple of fixes

### PR DESCRIPTION
**Pull request by eapearson**
- Takes care of https://kbase-jira.atlassian.net/browse/TASK-378, ensuring that sub-input widgets properly enforce and display the "required" constraint.
- Fixes label-display in report viewer
- Fixes import job panel not updating job status

### CODE REVIEW

**Comment by eapearson** [comment]
hold off for a sec, I have another fix coming.

**Comment by eapearson** [comment]
This should be ready to go. The build was failing for a while when CI was in distress.

File: kbase-extension/static/kbase/js/widgets/appWidgets2/fieldWidgetMicro.js [hunk 1]
```diff
@@ -0,0 +1,561 @@
...
+
+        function parameterInfoTypeRules(spec) {
+            switch (spec.data.type) {
+            case 'float':
+                return [
+                    tr([th('Min'), td(spec.data.constraints.min)]), // update this in the spec
+                    tr([th('Max'), td(spec.data.constraints.max)])
+                ];
+            case 'int':
+                // just for now ...
```
Comment (briehl) [inline kbase-extension/static/kbase/js/widgets/appWidgets2/fieldWidgetMicro.js]: Are 'int' and 'float' expected to be treated the same here? If so, should just merge into the same case.

File: kbase-extension/static/kbase/js/widgets/appWidgets2/fieldWidgetMicro.js [hunk 2]
```diff
@@ -0,0 +1,561 @@
...
+        }
+
+        function parameterInfoRules(spec) {
+            return table({ class: 'table table-striped' }, [
+                tr([th('Required'), td(spec.data.constraints.required ? 'yes' : 'no')]),
+                tr([th('Data type'), td(spec.data.type)]),
+                // tr([th('Field type'), td(spec.spec.field_type)]),
+                tr([th('Multiple values?'), td(spec.multipleItems ? 'yes' : 'no')]),
+                (function () {
+                    //                    if (!spec.spec.default_values) {
```
Comment (briehl) [inline kbase-extension/static/kbase/js/widgets/appWidgets2/fieldWidgetMicro.js]: Could just remove commented code blocks.

File: kbase-extension/static/kbase/js/widgets/appWidgets2/input/sequenceInput.js [hunk 3]
```diff
@@ -170,33 +172,46 @@ define([
                         }),
                         // TODO: should be a very lightweight wrapper widget here,
                         // at least to create and manage the channel.
-                        inputWidget = InputWidget.make({
-                            bus: inputBus,
-                            paramsChannelName: config.paramsChannelName,
-                            channelName: inputBus.channelName,
+                        // inputWidget = InputWidget.make({
```
Comment (briehl) [inline kbase-extension/static/kbase/js/widgets/appWidgets2/input/sequenceInput.js]: Also, some removable commented code here.

**Comment by briehl** [comment]
Looks good now!

**Pull request merged**